docs/developer/payments/lifecycle.mdx

```diff
@@ -173,27 +258,126 @@ it may still remain unpaid if the pending amounts fail to process.
+- on 10 AM Saleor receives `CHARGE_SUCCESS` event from payment app that was processed at 9:50 AM by payment provider
+- on 10:20 AM Saleor receives `AUTHORIZATION_SUCCESS` event, that was process at 9:30 AM by payment provider.
+
+This will result in a following list of events:
+- `AUTHORIZATION_SUCCESS` with `date: 9:30 AM`
+- `CHARGE_SUCCESS` with `date: 9:50 AM`
+
+### Reporting new events
+
+#### Automatic events
+
+Events are created automatically by Saleor when calling `transactionInitialize` mutation. It creates either:
+- `AUTHORIZATION_REQUEST` when `action` in mutation input or `channel.transactionFlowStrategy` is `AUTHORIZATION`
+- or `CHARGE_REQUEST` when `action` in mutation input or `channel.transactionFlowStrategy` is `CHARGE`
+
+Events are also created when requesting action in Saleor (refund, charge, cancel) via [`transactionRequestAction`](/api-reference/payments/mutations/transaction-request-action.mdx) mutation.
+
+Such events don't have initially any `pspReference`, it's filled in once Saleor receives a response from payment app.
+
+Saleor also creates automatically `_FAILURE` events when an exception happens:
+- when handling `transactionEventReport` mutation
+- when handling incorrect response from [sync webhook sent to payment app](/developer/extending/webhooks/synchronous-events/transaction.mdx)
+
+Such events don't have `pspReference` and include a `message` that explains the error.
+
+#### App events
+
+Events are also created by a payment app that processes the payment:
+
+- When app sends response to any of the [Transaction synchronous events](/developer/extending/webhooks/synchronous-events/transaction.mdx)
+  - Saleor will either fill missing `pspReference` on the automatically created event that already exists
+  - or will create a new event if it has different `type` (e.g. user requested `AUTHORIZATION`, but app returned `CHARGE_SUCCESS` - we'll have two events `AUTHORIZATION_REQUEST` with no `pspReference` and `CHARGE_SUCCESS`)
+- When [`transactionEventReport` mutation is called](/developer/payments/transactions.mdx#reporting-actions-for-transactions) (usually happens when payment provider sends asynchronous webhook)
+
+### Rules of calculating amounts
+
+Each event impacts different amount on `TransactionItem`:
+
+:::caution
+Each event can be excluded from calculation if it has matching `_FAILURE` event with the same `pspReference` and newer `time`.
+:::
+
+- `authorizedAmount` - `AUTHORIZATION_SUCCESS` event amount
+  - there can be only one `AUTHORIZATION_SUCCESS` event for given TransactionItem
+  - it can be modified by `AUTHORIZATION_ADJUSTMENT` event
+  - subtracting `canceledAmount`
+- `authorizePendingAmount` - sum of `AUTHORIZATION_REQUEST` event amounts
+- `chargedAmount` - sum of `CHARGE_SUCCESS` event amounts
+  - and sum `REFUND_REVERSE` event amounts
+  - subtracting all `CHARGE_BACK` events that lower this amount
+  - subtracting `refundedAmount`
```

Comment:
IMHO it is not clear to me. It is not like that `refundedAmount` has impact on `chargedAmount`. It is more like:
- Having REFUND_SUCCESS event would subtract the `event.amount` from the calculated `chargedAmount` and add `event.amount` to refundedAmount. 
- Having pending REFUND_REQUEST event would subtract the `event.amount` from the calculated `chargedAmount` add `event.amount` to refundPendingAmount.

Reply:
hmm, ok, could we word this differently then? 🤔 

I think in context of `chargedAmount` it's actually easier to subtract `refundedAmount` and `refundPendingAmount` rather than: "if you have REFUND_SUCCESS, then it subtracts event's value from chargedAmount and puts it in refundedAmount".

Correct me if I'm wrong but this is also how this calculation works in Saleor's code, right? We don't operate on individual events but on sum (`refundedAmount`, `canceledAmount`, etc.)

Reply:
> Correct me if I'm wrong but this is also how this calculation works in Saleor's code, right? We don't operate on individual events but on sum (refundedAmount, canceledAmount, etc.)

We calculate the values based on the events that we have. We matched them by PSP reference and their type, and apply different action for amounts. 
The amounts like `chargedAmount`, `refundedAmount`, `chargePendingAmount` etc are the result of calculations created based on the events. 


> I think in context of chargedAmount it's actually easier to subtract refundedAmount and refundPendingAmount rather than: "if you have REFUND_SUCCESS, then it subtracts event's value from chargedAmount and puts it in refundedAmount".

If we want to keep this simply, then I guess it is okay, but I would suggest provide some different structure. I mean, having entry like: `- subtracting refundedAmount` , wasn't very clear for me. Personally I would expect to have info in section for `refundedAmount` that would say, that it reduces the value of `chargedAmount`, same for `refundPendingAmount`.

Reply:
Sure, I don't want to drag this any longer, we can do this in separate PR.

I personally find having global description talking about amounts easier to understand than reading entire events description, but if this is incorrect we should point out that Saleor itself doesn't operate on these amounts, but rather adds or subtracks from TransactionItem amounts while iterating over events? I don't know how to put it in words to be honest 😅 

btw. Our `REFUND_SUCCESS` section already describes this:

> If there is no related `REFUND_REQUEST` event, the provided `amount` will reduce `transaction.chargedAmount`.

Is this this right though? I think we decrease `chargedAmount` anyway? Or does it mean we decrease `chargedAmount` first on `REFUND_REQUEST`, so when there's matching `REFUND_SUCCESS` there's nothing to be changed?